### DISCUSSION TITLE
Adding a resolver with the same parentType as itself

### DISCUSSION THREAD

**Discussion by jeffcrow5** [Netflix/dgs-framework#1607] [Q&A] [answered]
I have a data fetcher that returns an EcclesiasticalUnit object. The EcclesiasticalUnit type has fields `EcclesiasticalUnit parentUnit` and `List<EcclesiasticalUnit> childUnits`. I'm trying to create two data fetchers that return each of these pieces of data from my data source, to allow for recursive calling of the EcclesiasticalUnit type. I've written the following already:
```
@DgsQuery
public EcclesiasticalUnit ecclesiasticalUnit(DgsDataFetchingEnvironment dfe, @InputArgument Long id) { ... }

@DgsData(parentType = "EcclesiasticalUnit")
public EcclesiasticalUnit parentUnit(DgsDataFetchingEnvironment dfe) { ... }

@DgsData(parentType = "EcclesiasticalUnit")
public List<EcclesiasticalUnit> childUnits(DgsDataFetchingEnvironment dfe) { ... }
```

The first method works and returns an EcclesiasticalUnit object as expected. However, when either of the other two data fetchers exist in the code, I get an error that says something like:
```
Factory method 'schema' threw exception with message: Parent type EcclesiasticalUnit not found, but it was referenced on com.netflix.graphql.dgs.internal.DgsSchemaProvider in @DgsData annotation for field parentUnit
```

It seems that it's not recognizing the EcclesiasticalUnit type for some reason. Am I doing something wrong with my configuration?

**Comment by srinivasankavitha** [comment]
Maybe check if there are typos and that the type name matches what is
defined in the schema?

**Reply by jeffcrow5** [reply]
I've confirmed that my type name and casing matches and that there are no typos. I also confirmed that "parentUnit" is the name of the field inside the EcclesiasticalUnit type. It seems to be only happen when the return type and parentType are the same.

**Comment by srinivasankavitha** [comment, marked as answer]
I just tried this scenario and it works fine for the schema below:
```
type Query {
   movies: [Movie]
}

type Movie {
    title: String
    director: String
    similar: Movie
    related: [Movie]
}
```

With data fetchers defined as 

```
 @DgsQuery
    public List<Movie> movies(DataFetchingEnvironment dfe) {
        List<Movie> movies = new ArrayList<>();
        ...
        return movies;
    }

    
    @DgsData(parentType = "Movie")
    public String director() {
        return "some director";
    }

    @DgsData(parentType = "Movie")
    public Movie similar() {
       ...
        return new ActionMovie();
    }

    @DgsData(parentType = "Movie")
    public List<Movie> related() {
        List<Movie> movies = new ArrayList<>();
        return movies;
    }

```

I think this is the scenario you have above and it works as expected. You could try running through a debugger to see what's going on with the data fetching.

**Comment by jeffcrow5** [comment]
After looking further into the issue, I realized that there was a problem with the specific implementation that I'm using. It ended up being unrelated to the fact that it was the same type as its parent. Thank you for the responses!
